UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -221,16 +224,25 @@ public AgentInfo Info
         /// becomes enabled or active.
         void OnEnable()
         {
-            m_Id = gameObject.GetInstanceID();
-            OnEnableHelper();
-
-            m_Recorder = GetComponent<DemonstrationRecorder>();
+            LazyInitialize();
         }
 
         /// Helper method for the <see cref="OnEnable"/> event, created to
         /// facilitate testing.
-        void OnEnableHelper()
+        public void LazyInitialize()
```

Comment:
Should this really be public ?

Reply:
It needs to be public - internal won't cut it since the ModelOverrider calls it, and that's in the MLAgentsExamples namespace.